precompiles/dapps-staking/src/tests.rs

```diff
@@ -0,0 +1,598 @@
+use crate::mock::{
+    advance_to_era, default_context, evm_call, exit_error, initialize_first_block,
+    precompile_address, Call, EraIndex, ExternalityBuilder, Origin, Precompiles, TestAccount, AST,
+    BLOCK_REWARD, UNBONDING_PERIOD, *,
+};
+use fp_evm::PrecompileOutput;
+use frame_support::{assert_ok, dispatch::Dispatchable};
+use pallet_evm::{ExitSucceed, PrecompileSet};
+use sha3::{Digest, Keccak256};
+use sp_core::H160;
+use sp_runtime::Perbill;
+use std::collections::BTreeMap;
+
+use crate::utils;
+
+#[test]
+fn selector_out_of_bounds_nok() {
+    ExternalityBuilder::default().build().execute_with(|| {
+        initialize_first_block();
+
+        // Use 3 bytes selector. 4 bytes are needed
+        let selector_nok = vec![0x01, 0x02, 0x03];
+
+        let expected = Some(Err(exit_error("Selector too short")));
+
+        assert_eq!(
+            Precompiles::execute(
+                precompile_address(),
+                &selector_nok,
+                None,
+                &default_context(),
+            ),
+            expected
+        );
+    });
+}
+#[test]
+fn selector_unknown_nok() {
+    ExternalityBuilder::default().build().execute_with(|| {
+        initialize_first_block();
+
+        // We use 3 bytes selector. 4 byts are needed
+        let selector_nok = vec![0x01, 0x02, 0x03, 0x04];
+
+        let expected = Some(Err(exit_error("No method at given selector")));
+
+        assert_eq!(
+            Precompiles::execute(
+                precompile_address(),
+                &selector_nok,
+                None,
+                &default_context(),
+            ),
+            expected
+        );
+    });
+}
+
+#[test]
+fn current_era_is_ok() {
+    ExternalityBuilder::default().build().execute_with(|| {
+        initialize_first_block();
+
+        let selector = &Keccak256::digest(b"current_era()")[0..4];
```

Comment:
Can this name be fetched from the Call object perhaps?

Reply:
@Dinonard I don't think there is Call object for this, since this string is from Solidity contract. Please see
https://docs.soliditylang.org/en/v0.8.10/abi-spec.html#function-selector

Reply:
Oh I see.